### ISSUE TITLE
PHP Warning when there are no internal forms

### ISSUE THREAD

**Issue by ictbeheer** [CalderaWP/Caldera-Forms#2569]
## What Version Of Caldera Forms, WordPress and PHP Are You Using?
WordPress Version: 4.9.6PHP Version: 7.2.5MySQL Version: 5.5.5Caldera Forms Version: 1.7.0WP_DEBUG: 1

## Does Your Issue Persist When You Disable All Other Plugins and Switch To The Default Theme?
Yes

## What Is The Unexpected Behaviour?
When there a no internal forms (for example when all forms are defined in PHP), a PHP warning occurs.

## What PHP Errors Have You Logged While Reproducing This Bug?
`[01-Jun-2018 18:11:48 UTC] PHP Warning:  in_array() expects parameter 2 to be array, null given in C:\xampp\htdocs\local\wp-content\plugins\caldera-forms\classes\forms.php on line 721`

## What JavaScript Errors Have You Seen While Reproducing This Bug?
None

## Possible solution
Make get_stored_forms return an empty array when there are no stored forms instead of null

**Comment by Shelob9** [comment]
Thanks for the report. #2575 is a duplicate of this.

1df8a54 fixes this issue.

**Comment by ictbeheer** [comment]
@Shelob9 unfortunately the warning still appears after updating to 1.7.2

Maybe the full stacktrace is helpful:
```
PHP Warning:  in_array() expects parameter 2 to be array, null given in C:\xampp\htdocs\local\wp-content\plugins\caldera-forms\classes\forms.php on line 725
PHP Stack trace:
PHP   1. {main}() C:\xampp\htdocs\local\wp-admin\admin-ajax.php:0
PHP   2. require_once() C:\xampp\htdocs\local\wp-admin\admin-ajax.php:22
PHP   3. require_once() C:\xampp\htdocs\local\wp-load.php:37
PHP   4. require_once() C:\xampp\htdocs\local\wp-config.php:123
PHP   5. do_action() C:\xampp\htdocs\local\wp-settings.php:450
PHP   6. WP_Hook->do_action() C:\xampp\htdocs\local\wp-includes\plugin.php:453
PHP   7. WP_Hook->apply_filters() C:\xampp\htdocs\local\wp-includes\class-wp-hook.php:310
PHP   8. Caldera_Forms_GDPR::register_gdpr() C:\xampp\htdocs\local\wp-includes\class-wp-hook.php:286
PHP   9. Caldera_Forms_GDPR::enabled_forms() C:\xampp\htdocs\local\wp-content\plugins\caldera-forms\classes\gdpr.php:28
PHP  10. Caldera_Forms_Forms::get_form() C:\xampp\htdocs\local\wp-content\plugins\caldera-forms\classes\gdpr.php:124
PHP  11. Caldera_Forms_Forms::is_internal_form() C:\xampp\htdocs\local\wp-content\plugins\caldera-forms\classes\forms.php:102
PHP  12. in_array() C:\xampp\htdocs\local\wp-content\plugins\caldera-forms\classes\forms.php:725
```

**Comment by techteamCarvajal** [comment]
someone know how to solve it 
Warning: in_array() expects parameter 2 to be array, null given in /home4/oburgos68/public_html/wp-content/plugins/caldera-forms/classes/forms.php on line 725

**Comment by techteamCarvajal** [comment]
Sorry someone, knows how to solve it, I updated caldera and know i cant solve it

Warning: in_array() expects parameter 2 to be array, null given in /caldera-forms/classes/forms.php on line 725

**Comment by Shelob9** [comment]
#2736 tracks this bug fix for CF 1.8.0
